sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/SerializerEncoding.java

```diff
@@ -97,4 +80,43 @@ public static SerializerEncoding fromHeaders(HttpHeaders headers) {
 
         return JSON;
     }
+
+    /*
+     * There is a limited set of serialization encodings that are known ahead of time. Instead of using a TreeMap with
+     * a case-insensitive comparator, use an optimized search specifically for the known encodings.
+     */
+    private static SerializerEncoding checkForKnownEncoding(String contentType) {
```

Comment:
do we have a sufficient test coverage for it?

Reply:
Yep, there is a whole test suite for this https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/test/java/com/azure/core/util/serializer/SerializerEncodingTests.java